binary/script/build_and_pr.sh

```diff
@@ -43,7 +42,9 @@ function check_github_token() {
 function changes_detected() {
     local file="${1:-sha512sum.txt}"
 
-    if sha512sum -c "${file}"; then
+    create_checksums "${file}"
```

Comment:
This feels like it changes the nature of this function. it no longer just detects changes but creates them. Would you be open to moving this out of here and maybe down to func main?

Reply:
I can definitely refactor this.

Reply:
Nevermind, I did that wrong.

Reply:
Fixed.